tf_slim/__init__.py

```diff
@@ -48,4 +48,8 @@
 from tensorflow.python.util.all_util import make_all  # pylint:disable=g-direct-tensorflow-import
 # pylint: enable=unused-import,line-too-long,g-importing-member,wildcard-import
 
+
+from tensorflow.compat.v1 import disable_eager_execution
+disable_eager_execution()
```

Comment:
Do we need to disable_eager_execution in the init? or is enough in the tests?

Reply:
It's enough in some (though nearly all) tests. [Here is the change](https://github.com/adrianc-a/tf-slim/tree/no_eager_in_tests). Let me know if I should merge it into this branch.

Reply:
We want to avoid calling disable_eager_execution within libraries, because it can silently leak and cause issues. (Plus it's weird to have an import change your TensorFlow runtime settings.

My recommendation would be to create a utility method that looks something like:
```
def error_if_eager_is_enabled():
  with tf.init_scope():
    if tf.executing_eagerly():
      raise RuntimeError('TF-slim layers cannot be used when eager execution is enabled. Make sure to disable eager execution by calling `tf.compat.v1.disable_eager_execution()` or by using a custom estimator model_fn.')
```

And then run it at the start of every layer.
Then, all the tests should make sure to disable eager execution (but there could also be one test that makes sure the error gets raised if eager hasn't been disabled).